File: Vasya's calender/vasyas_calender.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;

int Vasyas_Calendar()
{ ll day_limit ;
  cin>>day_limit;
  ll months;
  cin>>months;
  ll days_in_months[months];
  
  for(ll i = 0;i<months;i++)
    cin>>days_in_months[i];
  
  ll ongoing_day = days_in_months[0] + 1 ;
  if(days_in_months[0] == day_limit)
      ongoing_day = 1 ;
  ll increase = 0;
  
  
  for(ll m = 1;m < months;m++)
  {
      if(ongoing_day != 1)
      { increase += day_limit - ongoing_day + 1 ;
        ongoing_day = 1 ;
      }
      ongoing_day += days_in_months[m];
  }
  
  return (increase);
}


int main() {
	int remained_test_cases = 1 ;
// 	cin>>remained_test_cases;
	while(remained_test_cases > 0)
	{   cout<<Vasyas_Calendar();
	    remained_test_cases--;
	}
	return 0;
}

File: Vasya's calender/vasyas_calender.py
def Vasyas_Calendar():
    days_limit = int(input())
    months = int(input())
    days_in_month = list(map(int,input().split()))
    ongoing_day = days_in_month[0] + 1 
    
    if days_in_month[0] == days_limit:
        ongoing_day = 1 
    increase = 0    
    
    for m in range(1,months):
        if ongoing_day != 1:
            # print("ongoing_day",ongoing_day)
            increase += days_limit + 1 - ongoing_day 
            # print(days_limit + 1 - ongoing_day)
            ongoing_day = 1 
        ongoing_day += days_in_month[m]  
    
    return increase    
            
            
        
remained_test_cases = 1 
# remained_test_cases = int(input())
while remained_test_cases > 0:
    print( Vasyas_Calendar())
    remained_test_cases -= 1 